spec/ddtrace/logger_spec.rb

```diff
@@ -0,0 +1,263 @@
+      end
+    end
+
+    context '#debug()' do
+      context 'default log level' do
+        before(:each) { log_debug() }
+        it { expect(lines.length).to eq(0) if lines.respond_to?(:length) }
+      end
+
+      context 'debug log level' do
+        let(:log_level) { Logger::DEBUG }
+
+        before(:each) { log_debug() }
+
+        it { expect(lines.length).to eq(1) if lines.respond_to?(:length) }
+        # We add traceback information to the log line for debug
+        it do
+          expect(lines[0]).to match(Regexp.new('D, \[[0-9:T.-]+ #[0-9]+\] DEBUG -- ddtrace: \[ddtrace\] ' \
+                                               '\([\/a-z_]+\.rb:[0-9]+:in `log_debug\'\) debug message\n'))
+        end
+      end
+    end
+
+    context '#error()' do
+      context 'default log level' do
+        before(:each) { log_error() }
+        it { expect(lines.length).to eq(1) if lines.respond_to?(:length) }
+        # We add traceback information to the log line for errors
+        it do
+          expect(lines[0]).to match(Regexp.new('E, \[[0-9:T.-]+ #[0-9]+\] ERROR -- ddtrace: \[ddtrace\] '\
+                                               '\([\/a-z_]+\.rb:[0-9]+:in `log_error\'\) error message\n'))
+        end
+      end
+    end
+
+    context '#warn' do
+      context '()' do
+        it { expect { Datadog::Tracer.log.warn('warn message') }.to_not raise_error }
+      end
+
+      context '() { message }' do
+        it { expect { Datadog::Tracer.log.warn() { 'warn message' } }.to_not raise_error }
+      end
+
+      context '(progname) { message }' do
+        it { expect { Datadog::Tracer.log.warn('bar') { 'warn message' } }.to_not raise_error }
+      end
+
+      context 'default log level' do
+        before(:each) { log_warn() }
+        it { expect(lines.length).to eq(1) if lines.respond_to?(:length) }
+        it { expect(lines[0]).to match(/W, \[[0-9:T.-]+ #[0-9]+\]  WARN -- ddtrace: \[ddtrace\] warn message\n/) }
+      end
+
+      context 'debug log level' do
+        let(:log_level) { Logger::DEBUG }
+        before(:each) { log_warn() }
+
+        it { expect(lines.length).to eq(1) if lines.respond_to?(:length) }
+        # We add traceback information to the log line for errors
+        it do
+          expect(lines[0]).to match(Regexp.new('W, \[[0-9:T.-]+ #[0-9]+\]  WARN -- ddtrace: \[ddtrace\] '\
+                                               '\([\/a-z_]+\.rb:[0-9]+:in `log_warn\'\) warn message\n'))
+        end
+      end
+    end
+
+    context 'debug_logging enabled' do
+      after(:each) { Datadog::Tracer.debug_logging = false }
+
+      it do
+        Datadog::Tracer.debug_logging = true
+        expect(Datadog::Tracer.log.level).to eq(Logger::DEBUG)
+
+        Datadog::Tracer.debug_logging = false
+        expect(Datadog::Tracer.log.level).to eq(Logger::WARN)
+      end
+    end
+  end
+
+  describe 'custom logger' do
+    let(:logger) do
+      logger = Logger.new(buf)
```

Comment:
You can do the following instead, which is more Ruby idiomatic. `tap` will return the Logger instance:

```
Logger.new(buf).tap do |logger|
  logger.level = Logger::INFO
end
```